File: sdkwork-backend-sdk-flutter/lib/src/api/invitation_relation.dart
import '../http/client.dart';
import '../models.dart';

class InvitationRelationApi {
  final HttpClient _client;
  
  InvitationRelationApi(this._client);

  /// Update an invitation relation
  Future<PlusApiResultInvitationRelationVO?> update(InvitationRelationForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/invitation/relation'), body: body, contentType: 'application/json');
    return response is PlusApiResultInvitationRelationVO ? response : null;
  }

  /// Create a new invitation relation
  Future<PlusApiResultInvitationRelationVO?> create(InvitationRelationForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/invitation/relation'), body: body, contentType: 'application/json');
    return response is PlusApiResultInvitationRelationVO ? response : null;
  }

  /// Get invitation relations by page
  Future<PlusApiResultPageInvitationRelationVO?> listByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/invitation/relation/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPageInvitationRelationVO ? response : null;
  }

  /// Get all invitation relations
  Future<PlusApiResultListInvitationRelationVO?> listAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/invitation/relation/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListInvitationRelationVO ? response : null;
  }

  /// Get an invitation relation by ID
  Future<PlusApiResultInvitationRelationVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/invitation/relation/${id}'));
    return response is PlusApiResultInvitationRelationVO ? response : null;
  }

  /// Delete an invitation relation
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/invitation/relation/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}

File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/invitation_relation.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import InvitationRelationForm, PlusApiResultBoolean, PlusApiResultInvitationRelationVO, PlusApiResultListInvitationRelationVO, PlusApiResultPageInvitationRelationVO, QueryListForm

class InvitationRelationApi:
    """invitation_relation API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: InvitationRelationForm) -> PlusApiResultInvitationRelationVO:
        """Update an invitation relation"""
        return self._client.put(f"/backend/v3/api/invitation/relation", json=body)

    def create(self, body: InvitationRelationForm) -> PlusApiResultInvitationRelationVO:
        """Create a new invitation relation"""
        return self._client.post(f"/backend/v3/api/invitation/relation", json=body)

    def list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPageInvitationRelationVO:
        """Get invitation relations by page"""
        return self._client.post(f"/backend/v3/api/invitation/relation/list", json=body, params=params)

    def list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListInvitationRelationVO:
        """Get all invitation relations"""
        return self._client.post(f"/backend/v3/api/invitation/relation/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultInvitationRelationVO:
        """Get an invitation relation by ID"""
        return self._client.get(f"/backend/v3/api/invitation/relation/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """Delete an invitation relation"""
        return self._client.delete(f"/backend/v3/api/invitation/relation/{id}")
